src/txn/txn.c

```diff
@@ -1978,29 +1988,36 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     if (!readonly)
         WT_IGNORE_RET(__wt_cache_eviction_check(session, false, false, NULL));
 
-    return (0);
-
+    if (0) {
```

Comment:
Why this change is needed?

Reply:
I think it is better to avoid using if (0) if we can.

Reply:
This is mainly to avoid calling `__wt_session_gen_leave` twice once in the normal scenario and another in the error scenario.
Another approach to avoid using `if (0)` is to call the rolling back the transaction and other things if `ret != 0`

```
err:
    /*
     * Leave the commit generation after all the updates are processed and the snapshot is released.
     */
    if (!prepare)
        __wt_session_gen_leave(session, WT_GEN_TXN_COMMIT);

    if (ret != 0){
        if (cursor != NULL)
            WT_TRET(cursor->close(cursor));

        if (locked)
            __wt_readunlock(session, &txn_global->visibility_rwlock);

        /* Check for a failure after we can no longer fail. */
        if (cannot_fail)
            WT_RET_PANIC(session, ret,
              "failed to commit a transaction after data corruption point, failing the system");

        /*
         * Check for a prepared transaction, and quit: we can't ignore the error and we can't roll
         * back a prepared transaction.
         */
        if (prepare)
            WT_RET_PANIC(session, ret, "failed to commit prepared transaction, failing the system");

        WT_TRET(__wt_session_reset_cursors(session, false));
        WT_TRET(__wt_txn_rollback(session, cfg));
    }
    return (ret);
```

Reply:
I actually would prefer calling `__wt_session_gen_leave ` twice. But this is mainly a personal preference.